### PR TITLE
improve startup performance for clojure apps

### PR DESCRIPTION
**Pull request by richardstartin**
This currently actually _increases_ the startup time for clojure koans on leiningen by 3 seconds (3s baseline, 13s on master, 16s on this branch, 37s prior to #1642).

### CODE REVIEW

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/GlobalIgnoresMatcher.java [hunk 1]
```diff
@@ -47,129 +47,145 @@ private GlobalIgnoresMatcher(final boolean skipAdditionalLibraryMatcher) {
...
-        || name.startsWith("com.jloadtrace.")
-        || name.startsWith("com.appdynamics.")
-        || name.startsWith("com.singularity.")
-        || name.startsWith("com.jinspired.")
-        || name.startsWith("org.jinspired.")) {
-      return true;
+    switch (name.charAt(0) - 'a') {
+        // starting at zero to get a tableswitch from javac, though it looks horrendous
+      case 'a' - 'a':
+      case 'b' - 'a':
```
Comment (bantonsson) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/GlobalIgnoresMatcher.java:53]: Can't you do a break here as well as in the other non matching ones?

Comment (richardstartin) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/GlobalIgnoresMatcher.java:53]: Can't believe I missed that, will measure again.

**Comment by richardstartin** [comment]
As of the last push leiningen/clojure koans starts up in under 10s

**Comment by richardstartin** [comment]
We should rip this out and replace it with a decent trie implementation in the long term, but this is a tactical fix to improve performance in in the clojure applications short term.

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/GlobalIgnoresMatcher.java [hunk 2]
```diff
@@ -47,129 +47,159 @@ private GlobalIgnoresMatcher(final boolean skipAdditionalLibraryMatcher) {
...
-      // FIXME: We should remove this once
-      // https://github.com/raphw/byte-buddy/issues/558 is fixed
-      if (name.equals("datadog.trace.bootstrap.instrumentation.java.concurrent.RunnableWrapper")
-          || name.equals(
-              "datadog.trace.bootstrap.instrumentation.java.concurrent.CallableWrapper")) {
-        return false;
-      }
-      return true;
+    switch (name.charAt(0) - 'a') {
+        // starting at zero to get a tableswitch from javac, though it looks horrendous
```
Comment (tylerbenson) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/GlobalIgnoresMatcher.java:51]: what happens if the case doesn't start with `0`?

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/GlobalIgnoresMatcher.java [hunk 3]
```diff
@@ -47,129 +47,159 @@ private GlobalIgnoresMatcher(final boolean skipAdditionalLibraryMatcher) {
...
+      case 'e' - 'a':
+        break;
+      case 'f' - 'a':
+        break;
+      case 'g' - 'a':
+        break;
+      case 'h' - 'a':
+        break;
+      case 'i' - 'a':
+        break;
```
Comment (tylerbenson) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/GlobalIgnoresMatcher.java:106]: can these all fall through to the last break?  (does the break need to be duplicated?)

**Pull request merged**